libs/serialization/docs/index.rst

```diff
@@ -11,8 +11,18 @@
 serialization
 =============
 
-This module provides serialization primitives and support for many C++ standard
-library and Boost types.
+This module provides serialization primitives and support for all built-in types as well as all C++ Standard library collection
+and utility types. This list is extended by |hpx| vocabulary types with proper support for global reference counting.
+
+The purest form of serializing data is to copy the content of the payload bit by bit; however, this method is impractical
+for generic C++ types, which might be composed of more than just regular built-in types. Instead, |hpx|'s approach to serialization
+is derived from the Boost Serialization library, and is geared towards allowing the programmer of a given class explicit control
+and syntax of what to serialize. It is based on operator overloading of two special archive types that hold a buffer or stream
+to store the serialized data and is responsible for dispatching the serialization mechanism to the intrusive or non-intrusive version.
+The serialization process is recursive. Each member that needs to be serialized must be specified explicitly. The advantage of this approach
+is that the serialization code is written in C++ and leverages all necessary programming techniques. The generic, user-facing interface allows
+for effective application of the serialization process without obstructing the algorithms that need special code for packing and unpacking.
+It also allows for optimizations in the implementation of the archives.
```

Comment:
The only practical request I have is if you would mind reformatting the paragraphs to be at most 80 characters wide. Some editors can do this automatically as well.

Reply:
@msimberg  Will do. You also mentioned that I should include examples of serialization. Do you have any references you would recommend to help me find or build examples?